instrumentation/http/src/main/java/brave/http/HttpResponseParser.java

```diff
@@ -0,0 +1,153 @@
+     */
+    @Override public void parse(@Nullable HttpResponse response, @Nullable Throwable error,
+      TraceContext context, SpanCustomizer customizer) {
+      int statusCode = 0;
+      if (response != null) {
+        statusCode = response.statusCode();
+        String nameFromRoute = spanNameFromRoute(response, statusCode);
+        if (nameFromRoute != null) customizer.name(nameFromRoute);
+        String maybeStatus = maybeStatusAsString(statusCode, 299);
+        if (maybeStatus != null) customizer.tag("http.status_code", maybeStatus);
+      }
+      error(statusCode, error, context, customizer);
+    }
+
+    /** The intent of this is to by default add "http.status_code", when not a success code */
+    @Nullable String maybeStatusAsString(int statusCode, int upperRange) {
+      if (statusCode != 0 && (statusCode < 200 || statusCode > upperRange)) {
+        return String.valueOf(statusCode);
+      }
+      return null;
+    }
+
+    static String spanNameFromRoute(HttpResponse response, int statusCode) {
+      String method = response.method();
+      if (method == null) return null; // don't undo a valid name elsewhere
+      String route = response.route();
+      if (route == null) return null; // don't undo a valid name elsewhere
+      if (!"".equals(route)) return method + " " + route;
+      if (statusCode / 100 == 3) return method + " redirected";
+      if (statusCode == 404) return method + " not_found";
+      return null; // unexpected
+    }
+
+    /**
+     * Override to change what data from the HTTP error are parsed into the span modeling it. By
+     * default, this tags "error" as the the status code, if the error parameter was null and the
+     * HTTP status is below 1xx or above 3xx.
+     *
+     * <p>Note: Either the httpStatus can be zero or the error parameter null, but not both. This
+     * does not parse the error, as it is assumed that the {@link ErrorParser has done so prior}.
+     *
+     * <p>Conventionally associated with the tag key "error"
+     */
+    protected void error(int httpStatus, @Nullable Throwable error, TraceContext context,
+      SpanCustomizer customizer) {
+      if (error != null) return; // the call site used the ErrorParser
+
+      // Instrumentation error should not make span errors. We don't know the difference between a
+      // library being unable to get the http status and a bad status (0). We don't classify zero as
+      // error in case instrumentation cannot read the status. This prevents tagging every response as
+      // error.
+      if (httpStatus == 0) return;
+
+      // Unlike success path tagging, we only want to indicate something as error if it is not in a
+      // success range. 1xx-3xx are not errors. It is endpoint-specific if client codes like 404 are
+      // in fact errors. That's why this is overridable.
+
+      // 1xx, 2xx, and 3xx codes are not all valid, but the math is good enough vs drift and opinion
```

Comment:
It's not brave's job to validate HTTP, so this comment is a bit overkill

Reply:
you are reviewing old code, so I guess it is a good thing to move things around occasionally :)